clang/test/OpenMP/nvptx_target_parallel_reduction_codegen_tbaa_PR46146.cpp

```diff
@@ -36,14 +36,14 @@ void test() {
 // CHECK1-NEXT:    [[DYN_PTR_ADDR:%.*]] = alloca ptr, align 8
 // CHECK1-NEXT:    [[DOTZERO_ADDR:%.*]] = alloca i32, align 4
 // CHECK1-NEXT:    [[DOTTHREADID_TEMP_:%.*]] = alloca i32, align 4
-// CHECK1-NEXT:    store ptr [[DYN_PTR]], ptr [[DYN_PTR_ADDR]], align 8, !tbaa [[TBAA10:![0-9]+]]
```

Comment:
From what I've heard from @jsjodin the TBAA info was pending to be removed. There is no good way to have OMPIRBuilder generate the TBAA so it has been muted for reductions codegen, let me know if this change is acceptable.

Otherwise, we would have to look into callbacks or other methods to retain this information.

Reply:
Browsing through these, it seems we only loose TBAA for accesses of allocas. This is fine.